rocketpy/mathutils/function.py

```diff
@@ -544,18 +549,25 @@ def set_discrete(
         one_by_one : boolean, optional
             If True, evaluate Function in each sample point separately. If
             False, evaluates Function in vectorized form. Default is True.
+        mutate_self : boolean, optional
+            If True, the original Function object source will be replaced by
+            the new one. If False, the original Function object source will
+            remain unchanged, and the new one is simply returned.
+            Default is True.
 
         Returns
         -------
         self : Function
         """
-        if self.__dom_dim__ == 1:
+        func = deepcopy(self) if not mutate_self else self
+
+        if func.__dom_dim__ == 1:
             xs = np.linspace(lower, upper, samples)
-            ys = self.get_value(xs.tolist()) if one_by_one else self.get_value(xs)
-            self.set_source(np.concatenate(([xs], [ys])).transpose())
-            self.set_interpolation(interpolation)
-            self.set_extrapolation(extrapolation)
-        elif self.__dom_dim__ == 2:
+            ys = func.get_value(xs.tolist()) if one_by_one else func.get_value(xs)
+            func.set_source(np.concatenate(([xs], [ys])).transpose())
+            func.set_interpolation(interpolation)
+            func.set_extrapolation(extrapolation)
+        elif func.__dom_dim__ == 2:
```

Comment:
Seems like the `set_discrete` method only works for 1-D and 2-D Functions, but this behavior is not documented tho. IMO, this is a problem, the function will be returned as the same. Or is there anything that I'm missing here?

Also, what is the advantage of setting the `one_by_one` to False? 

Not on the scope of this PR tho. I'd suggest to open an issue to improve the method for N-D functions if you agree that it would be nice.

Reply:
You are absolutely correct, this method behavior for higher dimensional Functions is really bad: the Function stays the same.

I do think this should be fixed and I am working on it, but this could be technically considered a breaking change.